IdentityCore/src/cache/crypto/MSIDAssymetricKeyPair+StkJwkExtensions.m

```diff
@@ -0,0 +1,151 @@
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#import "MSIDAssymetricKeyPair+StkJwkExtensions.h"
+
+#import "MSIDJsonObject.h"
+#import "NSData+MSIDAssymetricKeyPairExtensions.h"
+
+@implementation MSIDAssymetricKeyPair (StkJwkExtensions)
+
+- (nullable NSString *)generateStkJwk {
+    NSString * modulus;
+    NSString * exponent;
+    if (![self extractRawKeys:&modulus exponent:&exponent])
+    {
+        return nil;
+    }
+    
+    return [self formatAsJsonWebKey:@"RSA" modulus:modulus exponent:exponent];
+}
+
+// MacOS supports exporting public keys as a DER ASN.1 sequence.
+// A stkjwk can be created by exporting the modulus and exponent based on DER ASN.1 standards.
+// Documentation on the byte structure of DER ASN.1 encoding for the purpose of modulus and exponent parsing for an RSA
+// key is as follows:
+//   [1 byte sequence tag] [3 byte longform sequenceLength]
+//     [1 byte integer indicator] [3 byte longform modulusLength] [modulusLength sized modulusValue]
+//     [1 byte integer indicator] [1 byte shortform exponentLength] [exponentLength sized exponentValue]
+// Notes on expecations of this data format:
+//   sequence tag and length: these are used for internal memory management of a DER ASN.1 data structure
+//                 modulus: For a 2048 bit RSA key, if the modulus is encoded as a positive number, a 256 byte public key will be
+//                          prefixed with an one empty byte. This empty byte prefix is for internal memory management, and can be simply
+//                          stripped off for the purposes of exporting the modulus.
+//                exponent: this is always expected to be "AQAB" for RSA public keys
+//   shortform vs longform: two different types of lengths (1 or two bytes) that describe a values are supported in DER ASN.1 encoding:
+//                          - lengths shorter then 127 are simply encoded as a 1 byte int
+//                          - lengths larger then 126 are 3 bytes long broken into two parts: [1 prefix byte set to 127][a 2 byte int]
+// More details on the DER ASN.1 encoding can be found at:
+// https://docs.microsoft.com/en-us/windows/win32/seccertenroll/about-sequence
+- (bool)extractRawKeys:(NSString **)modulus
+              exponent:(NSString **)exponent {
+    // TODO: Code Reviewers: Is there a recommended code pattern for @available for macOS and iOS?
```

Comment:
!fyi

Reply:
I think common obj-c lib should already be 10.12 and above so not sure if you still need it.

Reply:
@kevindiclemente I have an outstanding PR which allows to extract modulus and exponent from the SecKeyRef objects
 https://github.com/AzureAD/microsoft-authentication-library-common-for-objc/blob/8f3287f7aa4fd5fc4bfd0fb650d6ea36326fec58/IdentityCore/src/cache/crypto/MSIDAssymetricKeyPair.m 
I also have a class that allows us to create jwk and signed access token jwt .https://github.com/AzureAD/microsoft-authentication-library-common-for-objc/blob/20771e31c23d733a48f3d2808215470a3a74ce3a/IdentityCore/src/pop_manager/MSIDDevicePopManager.m. I believe both our PR's have the same functionality and we should sync up on how we can combine them.

Reply:
Happy to sync! I recognize the code in the first PR and for the second PR, based on how pop code looks, is the token used in createSignedAccessToken use more then once, then this code should probably move to be a member on the assemetrickeypair object, but if its onetime then it could stay as extension